### PR TITLE
HTS-1291 - fixed back links

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala [hunk 1]
```diff
@@ -107,49 +110,57 @@ class EmailController @Inject() (val helpToSaveService:          HelpToSaveServi
...
+      ): Future[Result] =
+        SelectEmailForm.selectEmailForm.bindFromRequest().fold(
+          withErrors ⇒ Ok(views.html.email.select_email(email, withErrors, backLink)),
+          _.newEmail.fold[Future[Result]]{
+            SeeOther(routes.EmailController.confirmEmail(crypto.encrypt(email)).url)
+          } { newEmail ⇒
+            val updatedSession = session.copy(pendingEmail   = Some(newEmail), confirmedEmail = None)
+
+            if (updatedSession =!= session) {
+              updateSessionAndReturnResult(session.copy(pendingEmail   = Some(newEmail), confirmedEmail = None),
```
Comment (sureshhmrc) [inline app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala]: `session.copy(pendingEmail   = Some(newEmail), confirmedEmail = None)` is this not same as `updatedSession` declared above ?

File: conf/app.routes [hunk 2]
```diff
@@ -72,4 +72,6 @@ GET        /account-home/close-account-are-you-sure         @uk.gov.hmrc.helptos
 GET        /enter-uk-bank-details                           @uk.gov.hmrc.helptosavefrontend.controllers.BankAccountController.getBankDetailsPage
 POST       /enter-uk-bank-details                           @uk.gov.hmrc.helptosavefrontend.controllers.BankAccountController.submitBankDetails
 
-GET        /check-details                                   @uk.gov.hmrc.helptosavefrontend.controllers.RegisterController.checkDetails
\ No newline at end of file
+GET        /check-details                                   @uk.gov.hmrc.helptosavefrontend.controllers.RegisterController.checkDetails
+GET        /change-bank-details                             @uk.gov.hmrc.helptosavefrontend.controllers.RegisterController.changeBankDetails
+GET        /change-email-address                            @uk.gov.hmrc.helptosavefrontend.controllers.RegisterController.changeEmail
```
Comment (sureshhmrc) [inline conf/app.routes]: this seems in-consistent as any changes to emails or bank-details should go to appropriate controllers

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala]: whoops. I missed that

File: conf/app.routes (hunk 2 above)
Comment (AndyHWChung) [reply conf/app.routes]: I wanted to keep it close to check-details as they're triggered from that page

Comment (AndyHWChung) [reply conf/app.routes]: the email controllers getting big enough as it is!

File: app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala (hunk 1 above)
Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosavefrontend/controllers/EmailController.scala]: fixed and pushed now

File: conf/app.routes (hunk 2 above)
Comment (sureshhmrc) [reply conf/app.routes]: hmm , in that case we will split email controller logic into two classes one for Digital and one for DE

**Pull request merged**